File: backend/src/main/java/be/backend/services/PaymentService.java
package be.backend.services;

import be.backend.enums.PremiumPlan;
import be.backend.model.dto.AdminPaymentDto;
import be.backend.model.dto.PaymentDto;
import be.backend.model.request.SePayWebhookRequest;
import be.backend.model.response.CreatePaymentResponse;
import be.backend.model.response.PageResponse;
import be.backend.model.response.PaymentStatusResponse;

import java.util.List;

public interface PaymentService {

    /** Creates a PENDING payment and returns VietQR + transfer details. */
    CreatePaymentResponse createPremiumPayment(Integer userId, PremiumPlan plan);

    /** Processes a SePay transfer notification; grants premium on a valid match. */
    void handleSePayWebhook(SePayWebhookRequest payload);

    /** Validates the "Authorization: Apikey <key>" header SePay sends. */
    boolean isValidApiKey(String authorizationHeader);

    /** Lists a user's payment history. */
    List<PaymentDto> getUserPayments(Integer userId);

    PaymentStatusResponse getPaymentStatus(Integer orderCode);

    // --- Admin Methods ---
    PageResponse<AdminPaymentDto> getAllPaymentsAdmin(int page, int size, String status, Integer userId, String planType, String search);
    AdminPaymentDto getPaymentDetailAdmin(Integer paymentId);
    AdminPaymentDto updatePaymentStatusAdmin(Integer paymentId, String newStatus);
}

File: frontend/src/services/paymentService.js
import apiClient from './api'

const paymentService = {
  // POST /api/v1/payments/premium
  // Backend uses @AuthenticationPrincipal to get userId from JWT token
  // Body: { plan: 'MONTHLY' | 'YEARLY' }
  createPremiumPayment: (plan) => {
    return apiClient.post('/payments/premium', { plan })
  },

  // GET /api/v1/payments/users/{userId}
  getPaymentByUser: (userId) => {
    return apiClient.get(`/payments/users/${userId}`)
  },

  // GET /api/v1/payments/{orderCode}/status
  getPaymentStatus: (orderCode) => {
    return apiClient.get(`/payments/${orderCode}/status`)
  }
}

export default paymentService
